### PR TITLE
[cherry-pick-7930][Improvement]support instance to definition

### PR DESCRIPTION
**Pull request by brave-lee**
support instance to definition

### CODE REVIEW

File: dolphinscheduler-dao/src/main/resources/sql/dolphinscheduler_mysql.sql [hunk 1]
```diff
@@ -533,7 +534,7 @@ CREATE TABLE `t_ds_process_task_relation` (
   `create_time` datetime NOT NULL COMMENT 'create time',
   `update_time` datetime NOT NULL COMMENT 'update time',
   PRIMARY KEY (`id`),
-  KEY `project_code_process_definition_code_index` (`project_code`,`process_definition_code`) USING BTREE
+  KEY `idx_code` (`project_code`,`process_definition_code`)
```
Comment (lenboo) [inline dolphinscheduler-dao/src/main/resources/sql/dolphinscheduler_mysql.sql:537]: Dose the postgre sql also need to be changed ?

**Comment by lenboo** [comment]
@JinyLeeChina 
What functions will this PR affect?
For example, workflow creation and editing. What other test points do you think?

**Comment by lenboo** [comment]
close #7286 #7815

**Review by caishunfeng — APPROVED** [review]
+1

File: dolphinscheduler-dao/src/main/resources/sql/dolphinscheduler_mysql.sql (hunk 1 above)
Comment (brave-lee) [reply dolphinscheduler-dao/src/main/resources/sql/dolphinscheduler_mysql.sql:537]: Yes，it need

**Comment by brave-lee** [comment]
> @JinyLeeChina What functions will this PR affect? For example, workflow creation and editing. What other test points do you think?

I think the modification of workflow instances should be tested more. And when saving tasks at the front end, it is better to set taskgroupid to the default value, whether it is the interface defined and modified by workflow, the interface modified by workflow instance, or the interface modified by task definition

**Pull request merged**